### PR TITLE
Module : memory leak fix when the syntax error happens.

### PR DESCRIPTION
**Pull request by akroshg**
We were no removing modulerecord from MessageQueue when the exception happens - fixed that.
We were printing the exception in the notifymoduleready callback, which is not needed.
Populating the description field in the case of error, so that correct error will be propagated in the event of exception (sometime we use message field, other times description).

### CODE REVIEW

**Comment by obastemur** [comment]
@akroshg is this related to https://github.com/Microsoft/ChakraCore/issues/2735 ?

**Comment by akroshg** [comment]
Fixing 1st issue in the #3356

**Comment by akroshg** [comment]
@boingoing please have a look.

File: bin/ch/WScriptJsrt.cpp [hunk 1]
```diff
@@ -1666,7 +1666,10 @@ JsErrorCode WScriptJsrt::NotifyModuleReadyCallback(_In_opt_ JsModuleRecord refer
             wprintf(_u("NotifyModuleReadyCallback(exception) %S\n"), fileName.GetString());
         }
 
-        PrintException(*fileName, JsErrorScriptException);
+        // No need to print - just consume the exception
+        JsValueRef exception;
+        ChakraRTInterface::JsGetAndClearException(&exception);
+        exception; // unused
```
Comment (obastemur) [inline bin/ch/WScriptJsrt.cpp]: What is this for?

**Comment by akroshg** [comment]
@obastemur, I haven't tested that

**Comment by obastemur** [comment]
LGTM

File: bin/ch/WScriptJsrt.cpp (hunk 1 above)
Comment (akroshg) [reply bin/ch/WScriptJsrt.cpp]: We don't need to print the exception but we need to clear the exception. Otherwise we will run-into other problem.

File: bin/ch/MessageQueue.h [hunk 2]
```diff
@@ -102,6 +102,18 @@ class SortedList
...
+    {
+        DListNode<T>* node = head;
+
+        while (node != nullptr)
+        {
+            fn(node->data);
+            node = node->next;
+        }
+    }
+
```
Comment (jianchun) [inline bin/ch/MessageQueue.h]: nit: This function doesn't "remove" anything, shouldn't have name `RemoveAll`.

Comment (akroshg) [reply bin/ch/MessageQueue.h]: hmm good point, I'll rename this.

File: lib/Runtime/Library/JavascriptError.cpp [hunk 3]
```diff
@@ -831,6 +831,7 @@ namespace Js
         if (cse->ei.bstrDescription)
         {
             value = JavascriptString::NewCopySz(cse->ei.bstrDescription, scriptContext);
+            JavascriptOperators::OP_SetProperty(error, PropertyIds::description, value, scriptContext);
```
Comment (agarwal-sandeep) [inline lib/Runtime/Library/JavascriptError.cpp]: >description [](start = 68, length = 11)

Isn't description IE specific?  where ever we use description instead of message seems like a compat issue?

Comment (akroshg) [reply lib/Runtime/Library/JavascriptError.cpp]: Yes this to fix the Edge/Jshost problem. In the Edge case we never show what kind of syntax error found in the module case. With the 'description' being populated we are showing that now. ( I was planning to add jshost test as well - with some fix on the full side)
